src/components/ModalBackupWords.js

```diff
@@ -96,59 +105,78 @@ class ModalBackupWords extends React.Component {
+      // So we set start to 0 and end to 4
+      if (optionsStartIndex < 0) {
+        optionsStartIndex = 0;
+        optionsEndIndex = optionsStartIndex + 4;
+      }
+
+      // If index is 22 or 23, endIndex would be greater than the max
+      // So we set to the max and decrease the startIndex
+      const maxIndex = wordsArray.length - 1;
+      if (optionsEndIndex > maxIndex) {
+        optionsEndIndex = maxIndex;
+        optionsStartIndex = maxIndex - 4;
+      }
+
+      const options = _.shuffle(
+        wordsArray.slice(optionsStartIndex, optionsEndIndex + 1)
+      );
+
+      return {
+        backupIndex: backupIndex + 1,
+        correctWord: word,
+        options,
+        done: false,
+        last: index === wordsToValidate.length - 1,
+      };
+    });
+    this.setState({ showValidation: true, validationSteps });
+  };
 
   /**
    * Called when user finishes the backup. We validate the backup and shows a success message, in case of success
    */
-  handleValidateBackup = () => {
-    if (this.state.chosenWords.length !== WORDS_VALIDATION) {
-      this.setState({ errorMessage: t`Invalid number of words` });
+  handleValidationStepOption = (option, validationStep) => {
+    validationStep.chosenOption = option;
+    if (option !== validationStep.correctWord) {
+      this.setState({
+        validationSteps: this.state.validationSteps,
+        errorMessage: (
+          <span>
+            <span>{t`Wrong word. Please double check the words you saved.`}</span>
+            <strong onClick={() => this.setState({ showValidation: false, errorMessage: '' })}>{t`Click here to start the process again.`}</strong>
+          </span>
+        ),
+      });
       return;
     }
-    let validationArray = this.props.words.split(' ');
-    for (const word of this.state.chosenWords) {
-      const wordIdx = validationArray.indexOf(word);
-      if (wordIdx === -1) {
-        this.setState({ errorMessage: t`Invalid sequence of words` });
-        return;
-      }
-      validationArray = validationArray.slice(wordIdx);
+    if (validationStep.last) {
+      this.props.validationSuccess();
+      return;
     }
-    this.setState({ errorMessage: '' });
-    this.props.validationSuccess();
-  }
-
-  /**
-   * Called when user removes an already chosen word from the backup validation (just remove the word from the array)
-   */
-  chosenWordRemoved = (word) => {
-    let chosenWords = this.state.chosenWords;
-    const idx = chosenWords.indexOf(word);
-    chosenWords.splice(idx, 1);
-    this.setState({ chosenWords });
-  }
-
-  /**
-   * Called when user clicks in a word on the backup validation (just add the word in the array)
-   */
-  validationWordClicked = (word) => {
-    let chosenWords = this.state.chosenWords;
-    chosenWords.push(word);
-    this.setState({ chosenWords });
-  }
+    validationStep.done = true;
+    this.setState({ validationSteps: this.state.validationSteps });
```

Comment:
The same comment of the last `setState`, it's just setting the same data, isn't it?

Reply:
Same of https://github.com/HathorNetwork/hathor-wallet/pull/156#discussion_r537501578.

Reply:
Resolved by https://github.com/HathorNetwork/hathor-wallet/pull/156/commits/e3958cb3d1836dbfe98dd12e9cb4c1e60cdfe566.